students/N0vA/lesson_04/mailroom2.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+#### Mailroom Project Part 2 ####
+
+import sys
+import pathlib
+
+# Create list of donors and their donation history
+database = {'Bill Gates': [2000000, 250000000], 
+            'Jeff Bezos': [2000000],
+            'Elon Musk': [50000000, 10000000], 
+            'Howard Schultz': [1000000], 
+            'Paul Allen': [450000000]}
+
+# Send a thank you
+def thank_you():
+    
+    print('Alright.  Which donor would you like to send a thank you card?')
+    
+    while True:
+        person = input('Enter their name here or type "list" to see a list of donors: ')
+        past_donor = False
+
+        # Print list option
+        if person.lower() == 'list':
+            past_donor = True
+            for per in database.keys():
+                print(per)
+
+        else:
+            for name in database.keys():
```

Comment:
This should not longer be a for loop. Dictionary look ups are very efficient.

Utilize `if database.get(name)` check in your logic. 

I would recommend code similar to this:
```
amount = input('...')
if database.get(name):
    database[name].append(amount)
else:
    database[name] = [amount]
```

as you can see it is much simpler than your current code.
```